internal/analytics/event/event.go

```diff
@@ -0,0 +1,44 @@
+package event
+
+import "github.com/ActiveState/cli/internal/output"
+
+// EventData is the data that is synchronized on disk (or forwarded to the state-svc server)
+type EventData struct {
+	Category    string `json:"category"`
+	Action      string `json:"action"`
+	Label       string `json:"label"`
+	ProjectName string `json:"project"`
+	Output      string `json:"output_type"`
+	UserID      string `json:"user_id"`
+}
+
+func New(category, action string, label, projectName, out, userID *string) EventData {
```

Comment:
Can we make the dimensions a separate struct? This all being flat feels messy. Too many arguments.

eg. `events.New(cat, act, lbl, events.Dimensions{})`

Outside of the graphql layers I'd like to see an evident distinction between event data and dimensions.